pkg/scheduler/DESIGN.md

```diff
@@ -0,0 +1,548 @@
+# Query Request Queue Design: Queue Splitting and Prioritization
+
+The `RequestQueue` subservice embedded into the scheduler process is responsible for
+all decisions regarding enqueuing and dequeuing of query requests.
+While the `RequestQueue`'s responsibilities are relatively broad, including management of
+querier-worker connection lifecycles and graceful startup/shutdown,
+the queuing logic is isolated into a "tree queue" structure and its associated queue algorithms.
+
+## Tree Queue: What and Why
+
+The "tree queue" structure serves the purpose of a discrete priority queue.
+The requests are split into many queues, each of which is located at a leaf node in the tree structure.
```

Comment:
I'm not married to the word "builds," but I do think `serves the purpose of` can be condensed into one word for conciseness. 
```suggestion
The "tree queue" structure builds a discrete priority queue.
Requests are split into many queues, each of which is located at a leaf node in the tree structure.
```